main.go

```diff
@@ -18,45 +22,130 @@ var posts = []post{{
 	Title: "YOusdkfkd",
 }}
 
-func postsHandler(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-
-	handlers := map[string]http.HandlerFunc{
-		"GET": func(w http.ResponseWriter, r *http.Request) {
-			resp, _ := json.Marshal(posts)
-			fmt.Fprintf(w, string(resp))
-		},
-		"POST": func(w http.ResponseWriter, r *http.Request) {
-			decoder := json.NewDecoder(r.Body)
-			var p post
-			if err := decoder.Decode(&p); err != nil {
-				panic(err)
-			}
-
-			id := len(posts) + 1
-			p.ID = id
-			posts = append(posts, p)
-			resp, _ := json.Marshal(p)
-			fmt.Fprintf(w, string(resp))
-		},
-		"DELETE": func(w http.ResponseWriter, r *http.Request) {
-			println(r.URL.Path)
-		},
-	}
-
-	handler, ok := handlers[r.Method]
-	if !ok {
-		fmt.Fprintf(w, "Method not supported")
+func find(id int) (int, error) {
+	for i, v := range posts {
+		if v.ID == id {
+			return i, nil
+		}
+	}
+
+	return 0, errors.New("Not found")
+}
+
+func getPosts(w http.ResponseWriter, r *http.Request) {
+	resp, err := json.Marshal(posts)
+	if err != nil {
+		panic(err)
```

Comment:
`panic` is usually reserved for exceptional circumstances, you could use something like [http.Error](https://golang.org/pkg/net/http/#Error) here to write the error to the response writer. In this case, because `posts` is a variable of our making, it was probably our fault, and maybe you want to just say "Internal Server Error" instead. Don't forget to return after using http.Error as well.
```suggestion
		http.Error(w, "internal server error", http.StatusInternalServerError)
		return
```